art/attacks/evasion/pe_malware_attack.py

```diff
@@ -0,0 +1,778 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+This module implements the following white-box attacks related to PE malware crafting:
+    1) Append based attacks (example paper link: https://arxiv.org/abs/1810.08280)
+    2) Section insertion attacks (example paper link: https://arxiv.org/abs/2008.07125)
+    3) Slack manipulation attacks (example paper link: https://arxiv.org/abs/1810.08280)
+    4) DOS Header Attacks (example paper link: https://arxiv.org/abs/1901.03583)
+"""
+
+import json
+import random
+import copy
+import sys
+
+from typing import Optional, Union, Tuple, List, Dict, TYPE_CHECKING
+
+import lief
+import numpy as np
+import tensorflow as tf
+
+from art.estimators.estimator import BaseEstimator, LossGradientsMixin
+from art.attacks.attack import EvasionAttack
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_NEURALNETWORK_TYPE
+
+
+class MalwareGD(EvasionAttack):
```

Comment:
Is this attack TensorFlow specific?
```suggestion
class MalwareGDTensorFlow(EvasionAttack):
```